test/frame.js

```diff
@@ -145,6 +145,28 @@ describe('Frame', () => {
         sinon.assert.called(handle);
       });
 
+      it("if the origin changes in the frame, it updates in the app", () => {
+        const event = {
+          data: {jsonrpc: '2.0'},
+          source: frame.iframe.contentWindow,
+          origin: frame.origin
+        };
+
+        frame.handleProviderMessage(event);
+
+        sinon.assert.calledWith(handle, event.data);
+
+        const event2 = {
+          data: {jsonrpc: '2.0'},
+          source: frame.iframe.contentWindow,
+          origin: 'test'
+        };
+
+        frame.handleProviderMessage(event2);
+
+        sinon.assert.calledWith(handle, event2.data);
```

Comment:
you don't actually have an assertion that checks that `frame.origin` is equal to `event2.origin`

Reply:
Updated